core/src/env/srml/srml_only/impls.rs

```diff
@@ -153,4 +157,56 @@ where
     fn dispatch_raw_call(data: &[u8]) {
         unsafe { sys::ext_dispatch_call(data.as_ptr() as u32, data.len() as u32) }
     }
+
+    fn call_invoke(
+        callee: <Self as EnvTypes>::AccountId,
+        gas: u64,
+        value: <Self as EnvTypes>::Balance,
+        input_data: &[u8],
+    ) -> Result<(), CallError> {
+        let callee = callee.encode();
+        let value = value.encode();
+        unsafe {
```

Comment:
I would appreciate to keep the bounds of `unsafe` code as low as possible.
For example in this case you could go with:
```rust
let result = unsafe {
    sys::ext_call(
        callee.as_ptr() as u32,
        callee.len() as u32,
        gas,
        value.as_ptr() as u32,
        value.len() as u32,
        input_data.as_ptr() as u32,
        input_data.len() as u32,
    )
};
if result != 0 {
    return Err(CallError)
}
Ok(())
```

Reply:
I would go further and suggest that at the `core` layer we have just a `call` method which contains this unsafe block. This is more consistent with the mostly 1-1 mapping to `sys` functions at this layer. 

Then we could move the `call_invoke` and `call_evaluate` up to `exec_env` in the model layer, which would call into this.

Reply:
I am fine with both approaches. Maybe you are right that it is a better fit in `ExecEnv` though.